### PR TITLE
signing: add signature client

### PR DESCRIPTION
**Pull request by CaioTeixeira95**
### What

This PR adds the `SignatureClient` interface and the `EnvSignatureClient` which implements the `SignatureClient` interface. It receives a Stellar Private Key from the environment.

### Why

This layer handles the Stellar transaction signature.

### Known limitations

N/A

#### PR Structure

- [x] It is not possible to break this PR down into smaller PRs.
- [x] This PR does not mix refactoring changes with feature changes.
- [x] This PR's title starts with name of package that is most changed in the PR, or `all` if the changes are broad or impact many packages.

#### Thoroughness

- [x] This PR adds tests for the new functionality or fixes.
- [x] All updated queries have been tested (refer to [this](https://stackoverflow.com/a/29163465) check if the data set returned by the updated query is expected to be same as the original one).

#### Release

- [x] This is not a breaking change.
- [x] This is ready to be tested in development.
- [ ] The new functionality is gated with a feature flag if this is not ready for production.

### CODE REVIEW

File: internal/signing/signature_client.go
Comment (marcelosalloum) [inline internal/signing/signature_client.go:24]: I recommend you to override the String() method, in order to prevent distributionAccountFull from ever being printed.

It should be overridden in the value and not the pointer, like in:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/d3912e908b6c1cea56e7aad4b79b18b7e2c583e4/internal/transactionsubmission/store/db_vault.go#L22-L24



Here are the tests that prove it should be implemented in the value. They'd not pass if the implementation was in the pointer:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/d3912e908b6c1cea56e7aad4b79b18b7e2c583e4/internal/transactionsubmission/store/db_vault_test.go#L32-L64

File: internal/signing/signature_client.go
Comment (marcelosalloum) [inline internal/signing/signature_client.go:18]: I'm assuming you'll eventually add SignFeeBumpStellarTransaction when needed.

File: internal/signing/signature_client_test.go
Comment (marcelosalloum) [inline internal/signing/signature_client_test.go:77]: This command gives better feedback when it fails:
```suggestion
	assert.NoError(t, err)
```

File: go.sum
Comment (marcelosalloum) [inline go.sum:1]: I'm confused in why go.mod was not updated and go.sum was. Did you execute `go mod tidy` here? Is this something that should have been done in a previous PR?

**Review by marcelosalloum — APPROVED** [review]
LGTM! Plz, feel free to merge it after addressing the comments below.

File: go.sum
Comment (CaioTeixeira95) [reply go.sum:1]: I'm confused too. These PRs were split into 3, but I don't know why it appeared on this PR. May I fix it in another PR that only cleans up the dependencies? I think it would be easier.

Comment (marcelosalloum) [reply go.sum:1]: Since you're merging this into the `validators` branch (and not develop or main), this should be fine as long as @daniel-burghardt agrees with it.

Comment (daniel-burghardt) [reply go.sum:1]: Since this is now pointing to `main` shall we just run a `go mod tidy` before merging?

**Review by daniel-burghardt — APPROVED** [review]
LGTM! 🔥

**Pull request merged**
